app/controllers/posts_controller.rb

```diff
@@ -23,6 +23,18 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @comments = Comment.where(post_id: @post.id)
+    @comments.destroy_all
+    @likes = Like.where(post_id: @post.id)
+    @likes.destroy_all
+    @post.destroy
+    current_user.postCount -= 1
+    flash[:success] = 'Post deleted'
+    redirect_to root_path
+  end
+
   private
```

Comment:
- [ ] Please kindly ensure that the PostsController uses CanCanCan for authorization.

**Hint:**
  - Place the below at the top of the class
`load_and_authorize_resource`